components/HttpsCoponent_/HttpsRequest_.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * HTTPS GET Example using plain Mbed TLS sockets
+ *
+ * Contacts the howsmyssl.com API via TLS v1.2 and reads a JSON
+ * response.
+ *
+ * Adapted from the ssl_client1 example in Mbed TLS.
+ *
+ * SPDX-FileCopyrightText: The Mbed TLS Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * SPDX-FileContributor: 2015-2022 Espressif Systems (Shanghai) CO LTD
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <time.h>
+#include <sys/time.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/event_groups.h"
+#include "esp_wifi.h"
+#include "esp_event.h"
+#include "esp_log.h"
+#include "esp_system.h"
+#include "esp_timer.h"
+#include "nvs_flash.h"
+#include "nvs.h"
+#include "protocol_examples_common.h"
+#include "esp_sntp.h"
+#include "esp_netif.h"
+#include "freertos/queue.h"
+#include "lwip/err.h"
+#include "lwip/sockets.h"
+#include "lwip/sys.h"
+#include "lwip/netdb.h"
+#include "lwip/dns.h"
+#include "esp_tls.h"
+#include "sdkconfig.h"
+#if CONFIG_MBEDTLS_CERTIFICATE_BUNDLE
+#include "esp_crt_bundle.h"
+#endif
+#include "time_sync.h"
+extern SemaphoreHandle_t GetResponseSemaphore;
+extern QueueHandle_t BufQueue1;
+#define WEB_PORT "443"
+
+char *WebServerAddress;
+char *Web_URL;
+char *HttpsBuf;
+#define SERVER_URL_MAX_SZ 1024
+static const char *TAG = "Https";
+static void https_request_task(void *pvparameters);
+TaskHandle_t xTaskHandlerHTTPS;
+#define TIME_PERIOD (86400000000ULL)
+
+extern const uint8_t server_root_cert_pem_start[] asm("_binary_server_root_cert_pem_start");
+extern const uint8_t server_root_cert_pem_end[] asm("_binary_server_root_cert_pem_end");
+
+extern const uint8_t local_server_cert_pem_start[] asm("_binary_local_server_cert_pem_start");
+extern const uint8_t local_server_cert_pem_end[] asm("_binary_local_server_cert_pem_end");
+
+#ifdef CONFIG_EXAMPLE_CLIENT_SESSION_TICKETS
+static esp_tls_client_session_t *tls_client_session = NULL;
+static bool save_client_session = false;
+#endif
+/**
+
```

Comment:
ommit first null line, add space after every star(*)

Reply:
I did it ,